mockgcp/mockspanner/admin/database.go

```diff
@@ -0,0 +1,157 @@
+	pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/spanner/admin/database/v1"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"k8s.io/klog/v2"
+)
+
+var _ pb.DatabaseAdminServer = &SpannerDatabaseV1{}
+
+type SpannerDatabaseV1 struct {
+	*MockService
+	pb.UnimplementedDatabaseAdminServer
+}
+
+type spannerDatabaseName struct {
+	Project      *projects.ProjectData
+	InstanceName string
+	DatabaseName string
+}
+
+func (s *SpannerDatabaseV1) GetDatabase(ctx context.Context, req *pb.GetDatabaseRequest) (*pb.Database, error) {
+	name, err := s.parseDatabaseName(req.GetName())
+	if err != nil {
+		return nil, err
+	}
+	fqn := name.String()
+
+	obj := &pb.Database{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+func (s *SpannerDatabaseV1) CreateDatabase(ctx context.Context, req *pb.CreateDatabaseRequest) (*longrunningpb.Operation, error) {
+	databaseName := databaseNameFromCreateStatement(req.GetCreateStatement())
+	fqn := req.GetParent() + "/databases/" + databaseName
+
+	obj := &pb.Database{}
+	obj.Name = fqn
+	obj.DatabaseDialect = req.DatabaseDialect
+	if obj.DatabaseDialect == pb.DatabaseDialect_DATABASE_DIALECT_UNSPECIFIED {
+		obj.DatabaseDialect = pb.DatabaseDialect_GOOGLE_STANDARD_SQL
+	}
+	obj.VersionRetentionPeriod = versionRetentionPeriodFromDDL(req.GetExtraStatements())
+	obj.EncryptionConfig = req.EncryptionConfig
+	obj.State = pb.Database_READY
+	now := timestamppb.Now()
+	obj.CreateTime = now
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	metadata := &pb.CreateDatabaseMetadata{
+		Database: fqn,
+	}
+	return s.operations.StartLRO(ctx, fqn, metadata, func() (proto.Message, error) {
+		// Many fields are not populated in the LRO result
+		result := proto.Clone(obj).(*pb.Database)
+		result.Name = fqn
+		return result, nil
+	})
+}
+
+func (s *SpannerDatabaseV1) DropDatabase(ctx context.Context, req *pb.DropDatabaseRequest) (*emptypb.Empty, error) {
+	name, err := s.parseDatabaseName(req.Database)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	oldObj := &pb.Database{}
+	if err := s.storage.Delete(ctx, fqn, oldObj); err != nil {
+		return nil, err
+	}
+
+	return &emptypb.Empty{}, nil
+}
+
+func (s *SpannerDatabaseV1) UpdateDatabaseDdl(ctx context.Context, req *pb.UpdateDatabaseDdlRequest) (*longrunningpb.Operation, error) {
+	name, err := s.parseDatabaseName(req.Database)
+	if err != nil {
+		return nil, err
+	}
+	fqn := name.String()
+
+	obj := &pb.Database{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		return nil, err
+	}
+
+	metadata := &pb.UpdateDatabaseDdlMetadata{
+		Database:   fqn,
+		Statements: req.Statements,
+	}
+	return s.operations.StartLRO(ctx, fqn, metadata, func() (proto.Message, error) {
+		// Many fields are not populated in the LRO result
```

Comment:
nit: set the createTime, updateTime

Reply:
There is no field for updateTime for this resource, but just double-checked that createTime is being set.

Reply:
yeah, it looks like the Database uses `Reconciling` (bool) to tell if the update succeeds or not.